### DISCUSSION TITLE
Followed peoples' avatars and headers disappeared when server migrated to S3

### DISCUSSION THREAD

**Discussion by ghost** [mastodon/mastodon#20752] [Q&A]
Hi,
I set up a Mastodon droplet recently, and initially didn't select to use s3 storage. I then followed a few people, and decided I quite liked it so I migrated to s3. Since then, any new people I follow and media I download is fine. However, the people I followed before migrating to s3 no longer show images for avatar or header. I tried unfollowing and refollowing them, but that didn't fix it.

How do I force my Mastodon server to refetch avatar and header images for those early people I followed, and store the images on S3?

Thanks a lot!

A Mastodon noob.

**Comment by JetForMe** [comment]
I had a similar experience, but I was able to successfully migrate from local storage to s3. Then a friend chastised me for not using Digital Ocean Spaces (they're quite a bit cheaper), so I migrated it all to that. Unfortunately, that second migration has resulted in numerous (but not all) remote avatar images being broken.
Nothing I do in the CLI seems to fix the issue. I've tried refreshing accounts, domains, and clearing the cache. Not sure what to do.

**Reply by electroponix** [reply]
I am currently setting up a space for an existing instance. This makes me think I should go ahead and do a custom domain/ngnx vhost, but I don't want that many variables tonight. Another thread talked about it needing to change the space to publicly readable, but I don't know about that. Hope you got your issue fixed.

**Reply by JetForMe** [reply]
I did. It turned out I had the config wrong (Mastodon has no documentation about S3 config variables), and it was inserting an extra path component, resulting in items being uploaded to one (incorrect) path, but URLs written for another (correct) path. Unfortunately, I don't remember exactly what I had wrong, but this is what ended up working (for Spaces):

```
S3_ENABLED=true
S3_BUCKET=<bucket name>
S3_REGION=sfo2

AWS_ACCESS_KEY_ID=<secret>
AWS_SECRET_ACCESS_KEY=<secret>
S3_PROTOCOL=https
S3_HOSTNAME=sfo2.digitaloceanspaces.com
S3_ENDPOINT=https://sfo2.digitaloceanspaces.com/
```

My instance name is the same as my bucket name, which might be part of why this works.

**Comment by huyfififi** [comment]
I encountered the same issue. I used local storage at first and migrated the media files to AWS S3, but the existing users' avatars and headers didn't load.
And I found this discussion [https://github.com/mastodon/mastodon/discussions/17827](https://github.com/mastodon/mastodon/discussions/17827)
```
~/mastodon$ docker exec -it mastodon-web-1 sh
$ tootctl accounts refresh --all --verbose
```
magically resolved my issue, refreshing avatars and headers. Hope it works for you, too.
